src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs

```diff
@@ -34,31 +34,49 @@ public static Client BuildClient(OperationGroup operationGroup)
             }
 
             string clientName = operationGroup.CSharpName();
-            List<Method> methods = new List<Method>();
-            List<Paging> pagingMethods = new List<Paging>();
+            Dictionary<string, (Operation Operation, Method Method)> operationMethods = new Dictionary<string, (Operation, Method)>();
             foreach (Operation operation in operationGroup.Operations)
             {
                 Method? method = BuildMethod(operation, clientName, clientParameters);
                 if (method != null)
                 {
-                    methods.Add(method);
-                    var pageable = operation.Extensions.GetValue<IDictionary<object, object>>("x-ms-pageable");
-                    if (pageable != null)
+                    operationMethods.Add(operation.Language.Default.Name, (operation, method));
+                }
+            }
+
+            List<Method> nextPageMethods = new List<Method>();
+            List<Paging> pagingMethods = new List<Paging>();
+            foreach ((string _, (Operation operation, Method method)) in operationMethods)
+            {
+                IDictionary<object, object>? pageable = operation.Extensions.GetValue<IDictionary<object, object>>("x-ms-pageable");
+                if (pageable != null)
+                {
+                    //TODO: Assuming operationName is in this operation group: https://github.com/Azure/autorest.modelerfour/issues/85
+                    string? operationName = pageable.GetValue<string>("operationName")?.Split('_').Last();
+                    (Operation NextOperation, Method NextMethod)? next =
+                        operationName != null && operationMethods.TryGetValue(operationName,
```

Comment:
Should we throw if the `operationName` is not found?

Reply:
I really, really want M4 to handle stuff like this. IMO, at our point in the process, we shouldn't need to validate the data we received. But, this is a weird case. Do we error for anything else related to the data right now?

Reply:
> I really, really want M4 to handle stuff like this.

Me too.

I'm just trying to figure out what would be the most confusing option for customers getting a new method generated or an error.

> Do we error for anything else related to the data right now?

We are, often not cleanly, but a parameter that's defined in path but not in parameter list would result in error.

Reply:
Thinking about it a bit more, we should probably **not** go with a design that 'massages' issues away. I do have insight from the PowerShell generator around errors. I only had about 3-4 errors that could happen. Whenever a customer hit one, they wouldn't look into the error. They either created an issue in GitHub or messaged us directly about it. Part of the problem was where the error occurred in the code, we didn't have all the data necessary to make a detailed, actionable error message. In this situation, I can at least make a message that has enough context for them to investigate. So, I'll add that now.

Reply:
Error added. Also created: https://github.com/Azure/autorest.testserver/issues/129